src/main/java/io/cryostat/Cryostat.java

```diff
@@ -99,7 +98,8 @@ public void start(Promise<Void> future) {
                         client.deployer().deploy(client.httpServer(), false),
                         client.deployer().deploy(client.webServer(), false),
                         client.deployer().deploy(client.messagingServer(), false),
-                        client.deployer().deploy(client.ruleProcessor(), true));
+                        client.deployer().deploy(client.ruleProcessor(), true),
+                        client.deployer().deploy(client.recordingMetadataManager(), true));
```

Comment:
I'm having trouble with what seems like concurrency issues. Sometimes, rules do not start on targets because the discovery notification comes in earlier than the `RuleProcessor` is deployed. I'm not sure why this doesn't happen on the main branch.

E.g.
```
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Outgoing WS message: {"meta":{"category":"TargetJvmDiscovery","type":{"type":"application","subType":"json"},"serverTime":1661373648},"message":{"event":{"kind":"FOUND","serviceRef":{"connectUrl":"service:jmx:rmi:///jndi/rmi://cryostat:9093/jmxrmi","alias":"es.andrewazor.demo.Main","labels":{},"annotations":{"platform":{},"cryostat":{"HOST":"cryostat","PORT":"9093","JAVA_MAIN":"es.andrewazor.demo.Main","REALM":"JDP"}}}}}}
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.net.web.WebServer Verticle
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.messaging.MessagingServer Verticle
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Max concurrent WebSocket connections: 2147483647
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.rules.RuleProcessor Verticle
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Deploying io.cryostat.recordings.RecordingMetadataManager Verticle
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.net.HttpServer Verticle [9da5909a-2d82-4ced-81ae-672acae5b148]
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.discovery.DiscoveryStorage Verticle [7c23bb74-5fb4-4b5e-83d0-f80a28ab01de]
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.messaging.MessagingServer Verticle [39aeaded-1a4b-4937-b6b8-a09b3f52b117]
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
INFO: Deployed io.cryostat.rules.RuleProcessor Verticle [fe02195f-2e3a-4a1d-8a3e-2a351712ec86]
Aug 24, 2022 8:40:48 PM io.cryostat.core.log.Logger info
```
All that is different, is that the `recordingMetadataManager` is now a worker verticle and is deployed alongside the other verticles. I was thinking to just start the `recordingMetadataManager` and the `ruleProcessor` before the `discoveryStorage`, but I'm not sure. Thoughts on this?

Reply:
The rule processor both listens for target discovery notifications and activates rules on those (all rules defined at the time), as well as listens for rule creations and activates those on all known targets (everything in the database at the time).

Anyway, enforcing a deployment ordering on the verticles is okay if it's really necessary. Conceptually each verticle should be an independent unit and loosely coupled to the others, but in practice we haven't implemented everything that way.

What match expression are you using for these rules that don't properly trigger? ie what kind of target do you have that is appearing so quickly that it comes up before the RuleProcessor can deploy?

Reply:
Using smoketest, in this particular instance above, the `vertx-fib-demo` that is on port 9093 is appearing too quickly, and my rule which just has `matchExpression=true` doesn't act on it. But I think I've had different targets not have rules run on them too, I can run more times to check.

Reply:
Interesting. Yea, I suppose that makes sense.

The verticle ordering should probably be something like this:

1. HttpServer
2. WebServer
3. MessagingServer
4. RecordingMetadataManager
5. RuleProcessor
6. DiscoveryStorage

I think a chained/serial startup sequence in this order should work.